### PR TITLE
feat(system-backup): add system backup backingimage test

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/5085

Create a Volume with the BackingImage
Create the system backup
Delete the Volume and the BackingImage
Restore the system backup
Both Volume and BackingImage will be restored.

### CODE REVIEW

File: manager/integration/tests/test_system_backup_restore.py [hunk 1]
```diff
@@ -112,6 +116,66 @@ def test_system_backup_and_restore_volume_with_data(client, volume_name, set_ran
...
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    assert backing_image.sourceType == BACKING_IMAGE_SOURCE_TYPE_RESTORE
+
+    restored_volume = client.by_id_volume(volume_name)
+    wait_for_volume_restoration_completed(client, volume_name)
+    wait_for_volume_detached(client, volume_name)
+
+    restored_volume.attach(hostId=host_id)
+    restored_volume = wait_for_volume_healthy(client, volume_name)
+
```
Comment (c3y1huang) [inline manager/integration/tests/test_system_backup_restore.py:178]: Should also check the data?

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_system_backup_restore.py (hunk 1 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_system_backup_restore.py:179]: Thanks for the advice
I think no, we can focus on the volume backup/restore with BackingImage
And for the data consistency
1. Volume already has its own test above
2. BackingImage use checksum to verified the data during backup/restore. If it is not consistency, BackingImage will be failed and so is the test.

So I think we can remove the data test in this test case to make it simple.

File: manager/integration/tests/test_system_backup_restore.py [hunk 2]
```diff
@@ -112,6 +117,66 @@ def test_system_backup_and_restore_volume_with_data(client, volume_name, set_ran
...
+    Given a backingimage
+    And a volume created with the backingimage
+    When system backup created
+    Then system backup in state Ready
+
+    When volume deleted
+    And backingimage deleted
+    And restore system backup
+
+    When restore system backup
```
Comment (c3y1huang) [inline manager/integration/tests/test_system_backup_restore.py:136]: `When` should be followed by `And` or `Then`.
```suggestion
    And restore system backup
```

File: manager/integration/tests/test_system_backup_restore.py (hunk 1 above)
Comment (c3y1huang) [reply manager/integration/tests/test_system_backup_restore.py:179]: I see. Can you add a comment to ensure @longhorn/qa is informed that the data integrity testing is covered somewhere else by looking at the case?

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_system_backup_restore.py (hunk 1 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_system_backup_restore.py:185]: Sure!
Added.

File: manager/integration/tests/test_system_backup_restore.py (hunk 2 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_system_backup_restore.py]: Fixed

File: manager/integration/tests/test_system_backup_restore.py [hunk 3]
```diff
@@ -112,6 +117,72 @@ def test_system_backup_and_restore_volume_with_data(client, volume_name, set_ran
...
+    Given a backingimage
+    And a volume created with the backingimage
+    When system backup created
+    Then system backup in state Ready
+
+    When volume deleted
+    And backingimage deleted
+    And restore system backup
+
+    And restore system backup
```
Comment (c3y1huang) [inline manager/integration/tests/test_system_backup_restore.py:142]: Sorry, bad eye, I just noticed these are duplicated steps.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_system_backup_restore.py (hunk 3 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_system_backup_restore.py]: me too, fixed.

**Review by c3y1huang — APPROVED** [review]

**Review by yangchiu — APPROVED** [review]
LGTM

**Comment by chriscchien** [comment]
@mergify backport v1.6.x

**Pull request merged**